### ISSUE TITLE
Minimum wait time between requests to avoid cloudflare?

### ISSUE THREAD

**Issue by jamesNWT** [gigobyte/HLTV#400]
Hi guys!

Does anyone here have any idea what rate couldflare will flag me as a malicious bot at? I am using this API to create a dataset for a machine learning project and will probably make a few thousand reqeuests in one run. I can self-throttle but I'd like to know how severely I'll have to slow it down and wait between requests.

Also the requests will all be sequential, the next request won't be sent untill the response from the previous request is recieved, if that matters.

**Comment by petz15** [comment]
I'm not 100% sure but I had a request around every 5 seconds which seemed to work fine. But to be sure I highly advise you to use a VPN! This way you can test a couple of intervals and not having to wait for the temporary ban to be over.

**Comment by jamesNWT** [comment]
Thanks. I'm using 3.5 seconds right now, and it seems to be working. I also added a catch for cloudflare errors that waits 5 minutes and then tries to call the request again, which I haven't needed but I think is a decent approach. Not sure what the wait time on that needs to be but 5 minutes is my starting point. 

I made this post because I had been getting flagged with (what I thought were) requests up to 10 seconds apart, but that may have just been buggy code actually not waiting or something.
